pkg/cmd/search/issues/issues.go

```diff
@@ -0,0 +1,155 @@
+				}
+			}
+			if c.Flags().Changed("no-assignee") && noAssignee {
+				opts.Query.Qualifiers.No = append(opts.Query.Qualifiers.No, "assignee")
+			}
+			if c.Flags().Changed("no-label") && noLabel {
+				opts.Query.Qualifiers.No = append(opts.Query.Qualifiers.No, "label")
+			}
+			if c.Flags().Changed("no-milestone") && noMilestone {
+				opts.Query.Qualifiers.No = append(opts.Query.Qualifiers.No, "milestone")
+			}
+			if c.Flags().Changed("no-project") && noProject {
+				opts.Query.Qualifiers.No = append(opts.Query.Qualifiers.No, "project")
+			}
+			opts.Query.Keywords = args
+			if runF != nil {
+				return runF(opts)
+			}
+			var err error
+			opts.Searcher, err = shared.Searcher(f)
+			if err != nil {
+				return err
+			}
+			return shared.SearchIssues(opts)
+		},
+	}
+
+	// Output flags
+	cmdutil.AddJSONFlags(cmd, &opts.Exporter, search.IssueFields)
+	cmd.Flags().BoolVarP(&opts.WebMode, "web", "w", false, "Open the search query in the web browser")
+
+	// Query parameter flags
+	cmd.Flags().IntVarP(&opts.Query.Limit, "limit", "L", 30, "Maximum number of results to fetch")
+	cmdutil.StringEnumFlag(cmd, &order, "order", "", "desc", []string{"asc", "desc"}, "Order of results returned, ignored unless '--sort' flag is specified")
+	cmdutil.StringEnumFlag(cmd, &sort, "sort", "", "best-match",
+		[]string{
+			"comments",
+			"created",
+			"interactions",
+			"reactions",
+			"reactions-+1",
+			"reactions--1",
+			"reactions-heart",
+			"reactions-smile",
+			"reactions-tada",
+			"reactions-thinking_face",
+			"updated",
+		}, "Sort fetched results")
+
+	// Query qualifier flags
+	cmd.Flags().BoolVar(&includePrs, "include-prs", false, "Include pull requests in results")
+	cmdutil.NilBoolFlag(cmd, &opts.Query.Qualifiers.Archived, "archived", "", "Restrict search to archived repositories")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Assignee, "assignee", "", "Filter by assignee")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Author, "author", "", "Filter by author")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Closed, "closed", "", "Filter on closed at `date`")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Commenter, "commenter", "", "Filter based on comments by `user`")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Comments, "comments", "", "Filter on `number` of comments")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Created, "created", "", "Filter based on created at `date`")
+	cmdutil.StringSliceEnumFlag(cmd, &opts.Query.Qualifiers.In, "match", "", nil, []string{"title", "body", "comments"}, "Restrict search to specific field of issue")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Interactions, "interactions", "", "Filter on `number` of reactions and comments")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Involves, "involves", "", "Filter based on involvement of `user`")
+	cmdutil.StringSliceEnumFlag(cmd, &opts.Query.Qualifiers.Is, "visibility", "", nil, []string{"public", "private"}, "Filter based on visibility")
```

Comment:
Let's accept `internal` visibility as well, since this is about the visibility of the repo, and in repo search we support `internal` too since ES does as well.